From 3e24e76dfaf7367e0790c22aa1e740f3b68d91a3 Mon Sep 17 00:00:00 2001
From: Dmitry Belyavskiy <dbelyavs@redhat.com>
Date: Mon, 21 Aug 2023 16:08:39 +0200
Subject: [PATCH 44/48] 0109-fips-Zeroize-out-in-fips-selftest.patch

Patch-name: 0109-fips-Zeroize-out-in-fips-selftest.patch
Patch-id: 109
---
 providers/fips/self_test.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/providers/fips/self_test.c b/providers/fips/self_test.c
index 062d9df84a..64107d054b 100644
--- a/providers/fips/self_test.c
+++ b/providers/fips/self_test.c
@@ -339,6 +339,7 @@ static int verify_integrity(OSSL_CORE_BIO *bio, OSSL_FUNC_BIO_read_ex_fn read_ex
         goto err;
     ret = 1;
 err:
+    OPENSSL_cleanse(out, sizeof(out));
     OSSL_SELF_TEST_onend(ev, ret);
     EVP_MAC_CTX_free(ctx);
     EVP_MAC_free(mac);
-- 
2.41.0

